### PR TITLE
Filters in clients

### PR DESCRIPTION
**Pull request by prodhan29**
-following feature is in this pr
 

- searchText filter
- gender filter

it will resolve this issues: #208 #209

### CODE REVIEW

**Comment by prodhan29** [comment]
@p-netm interface for the DropdownOption and default values for gender dropdown need to move some where else. please suggest a proper structure.

File: clients/core/src/containers/pages/Clients/List/index.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    constructor(props: ClientListProps) {
+        super(props);
+        this.state = defaultClientListState;
+    }
+
+    public async componentDidMount() {
+        this.getDataFromServer();
+    }
+
+    getDataFromServer = async () => {
```
Comment (p-netm) [inline clients/core/src/containers/pages/Clients/List/index.tsx]: lets add a docstring here, and any other functions that should have one but currently does not.

File: clients/core/src/containers/pages/Clients/List/index.tsx [hunk 2]
```diff
@@ -37,6 +40,49 @@ const ClientTable: React.FC<ClientTableProps> = ({ tableData }: ClientTableProps
     <OpenSRPTable {...{ data: tableData, tableColumns: useClientTableColumns() }} />
 );
 
+/** interface for the pagination props */
+interface PaginationProps {
+    initialState: any;
```
Comment (prodhan29) [inline clients/core/src/containers/pages/Clients/List/index.tsx]: @p-netm these are the required options that need to be passed. can you suggest what should be do except using `any` here

File: clients/core/src/configs/__mocks__/env.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@
 export const OPENSRP_API_BASE_URL = 'https://test.smartregister.org/opensrp/rest/';
 export type OPENSRP_API_BASE_URL = typeof OPENSRP_API_BASE_URL;
 /** the clients endpoint NOTE: does not end with / */
-export const OPENSRP_CLIENT_ENDPOINT = 'client/search';
+export const OPENSRP_CLIENT_ENDPOINT = 'client/searchByCriteria';
```
Comment (p-netm) [inline clients/core/src/configs/__mocks__/env.ts]: can we move this to the constants file,

**Comment by prodhan29** [comment]
@p-netm is this branch in a state to get merged with master ?, if yes, then please do it, so that i can apply those pagination changes to other branches. pagination issue still persist in [household-list](https://github.com/OpenSRP/opensrp-web/pull/146) branch

**Pull request merged**